engine/src/main/java/org/terasology/engine/persistence/typeHandling/extensionTypes/SimpleUriTypeHandler.java

```diff
@@ -0,0 +1,23 @@
+// Copyright 2022 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.engine.persistence.typeHandling.extensionTypes;
+
+import org.terasology.engine.core.SimpleUri;
+import org.terasology.persistence.typeHandling.StringRepresentationTypeHandler;
+
+public class SimpleUriTypeHandler extends StringRepresentationTypeHandler<SimpleUri> {
+
+    @Override
+    public String getAsString(SimpleUri uri) {
+        if (uri == null) {
+            return "";
+        }
+        return uri.toString();
+    }
+
+    @Override
+    public SimpleUri getFromString(String representation) {
+        return new SimpleUri(representation);
```

Comment:
Lets add a check of `uri.isValid()` here. It's a cheap check, and doing it here instead of leaving it until later will make it easier to identify what the source string is and where it came from.

As for what we do if it _doesn't_ pass the check... uh, if we think the deserialization code is trying to avoid throwing exceptions, I guess we log the error but still return the invalid SimpleUri instance?

Reply:
Looking at the implementation of `StringRepresentationTypeHandler` returning null would yield `Optional.empty`, which is our equivalent to "could not parse"... That probably better than returning an invlid `SimpleUri`...

https://github.com/MovingBlocks/Terasology/blob/b367a180d4e32cc0e573f6567f6db54987c092a3/subsystems/TypeHandlerLibrary/src/main/java/org/terasology/persistence/typeHandling/StringRepresentationTypeHandler.java#L22